### PR TITLE
Logs old script params use to the Deprecation Logger

### PR DESCRIPTION
**Pull request by colings86**
Relates to #16910

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/script/ScriptParameterParser.java [hunk 1]
```diff
@@ -202,7 +208,12 @@ public ScriptParameterValue getDefaultScriptParameterValue() {
     }
 
     public ScriptParameterValue getScriptParameterValue(String parameterName) {
-        return parameterValues.get(parameterName);
+        ScriptParameterValue scriptParameterValue = parameterValues.get(parameterName);
+        if (scriptParameterValue != null) {
+            deprecationLogger.deprecated("this from of the script api is deprecated and will be removed from the java api in the next "
```
Comment (dadoonet) [inline core/src/main/java/org/elasticsearch/script/ScriptParameterParser.java]: May be my lack of english. Do you mean "This method from the script API is deprecated"?

**Comment by dadoonet** [comment]
Left a comment (feel free to ignore). LGTM

File: core/src/main/java/org/elasticsearch/script/ScriptParameterParser.java (hunk 1 above)
Comment (colings86) [inline core/src/main/java/org/elasticsearch/script/ScriptParameterParser.java]: I mant to say "this form of the script API...". Will fix

**Comment by colings86** [comment]
@dadoonet thanks for the review :)

**Pull request merged**